### PR TITLE
Add checkbox to disable Science

### PR DESCRIPTION
**Pull request by Bentechy66**
Users who want privacy should be given an easy way to get it, and this option provides a quick and easy way to do that. Thanks to @BrandonDyer64 for his help bug fixing.

(Popup looks like the below image)
Popup Screenshot
(Checkbox ticked by default)

### CODE REVIEW

File: src/popup/index.ts [hunk 1]
```diff
@@ -24,3 +24,20 @@ chrome.storage.local.get("voted", ({ voted }) => {
 document.getElementById("resetID")!.addEventListener("click", () => {
 	chrome.storage.local.set({ scienceUUID: v4() });
 });
+
+const enableScience: HTMLInputElement = <HTMLInputElement> document.getElementById("enableScience")!;
+
+chrome.storage.sync.get('scienceEnabled', (result) => {
+		if (typeof result.scienceEnabled == "undefined") {
+			chrome.storage.sync.set({ scienceEnabled: true });
+			setTimeout(function() {enableScience.checked = true}, 100);
```
Comment (haykam821) [inline src/popup/index.ts]: Could you use a callback here instead of `setTimeout`?

Comment (Bentechy66) [reply src/popup/index.ts]: Updated 👍

**Review by bakcxoj — APPROVED** [review]
No viruses. Seems to be a good addition.

**Comment by 9hp** [comment]
+1 good idea

File: src/assets/static/popup.html [hunk 2]
```diff
@@ -36,7 +36,8 @@ <h1>
 					<span class="label">Science:</span>
 					<span class="info"
 						title="We collect anonymized data on how many votes you've given. To reset your grouping UUID, click this button."></span>
-					<button id="resetID">Reset Anonymous ID</button>
+					<button id="resetID">Reset Anonymous ID</button><br/>
+					<label for="enableScience" class="label">Enable Science: </label><input name="enableScience" id="enableScience" type="checkbox"></input>
```
Comment (haykam821) [inline src/assets/static/popup.html]: Can you add an info span after the label, describing what the checkbox does?

File: src/event/index.ts [hunk 3]
```diff
@@ -83,13 +83,17 @@ const connect = (scienceUUID: string, modhash: string) => {
...
 			voted.push(...toVote);
 			chrome.storage.local.set({ voted });
-			ws.send(
-				JSON.stringify({
-					type: "science",
-					uuid: scienceUUID,
-					upvoted: voted.length
-				})
-			);
+			chrome.storage.sync.get('scienceEnabled', (result) => {
```
Comment (haykam821) [inline src/event/index.ts]: I'm not familiar with the style used for this repo, but I think this should be double quoted.

**Review by haykam821 — CHANGES_REQUESTED** [review]

**Comment by hedgehog1029** [comment]
possible enhancement would be a little description of what is actually sent on the science channel - we might do that ourselves though

**Comment by Bentechy66** [comment]
"Add info for what you are disabling and center checkbox" should be the above commit message

**Comment by Bentechy66** [comment]
@hedgehog1029 doesn't the "info" message next to the "Reset Anonymous ID" button not already tell the user what is sent out via science? It only sends out the anonymized number of upvotes along with an anonymized UUID, right?

**Comment by odensc** [comment]
LGTM. I'll make a few changes after merge.

**Pull request merged**